go/internal/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,211 @@
+// It allows configuring how telemetry data (traces and metrics) is exported to an OpenTelemetry collector.
+//
+// Example usage:
+//
+//	config := glide.OpenTelemetryConfig{
+//	    Traces: &glide.OpenTelemetryTracesConfig{
+//	        Endpoint:         "http://localhost:4318/v1/traces",
+//	        SamplePercentage: 10, // Sample 10% of commands
+//	    },
+//	    FlushIntervalMs: &interval, // interval := int64(1000)
+//	}
+//	err := glide.GetInstance().Init(config)
+//	if err != nil {
+//	    log.Fatalf("Failed to initialize OpenTelemetry: %v", err)
+//	}
+type OpenTelemetryConfig struct {
+	// Traces configuration for exporting trace data. If nil, trace data will not be exported.
+	Traces *OpenTelemetryTracesConfig
+	// Metrics configuration for exporting metrics data. If nil, metrics data will not be exported.
+	Metrics *OpenTelemetryMetricsConfig
+	// FlushIntervalMs is the interval in milliseconds between consecutive exports of telemetry data.
+	// If nil, a default value will be used.
+	FlushIntervalMs *int64
+}
+
+// OpenTelemetryTracesConfig represents the configuration for exporting OpenTelemetry traces.
+//
+// - Endpoint: The endpoint to which trace data will be exported. Expected format:
+//   - For gRPC: `grpc://host:port`
+//   - For HTTP: `http://host:port` or `https://host:port`
+//   - For file exporter: `file:///absolute/path/to/folder/file.json`
+//   - SamplePercentage: The percentage of requests to sample and create a span for, used to measure command duration.
+//     Must be between 0 and 100. If not specified, defaults to 1.
+type OpenTelemetryTracesConfig struct {
+	Endpoint         string
+	SamplePercentage int32
+}
+
+// OpenTelemetryMetricsConfig represents the configuration for exporting OpenTelemetry metrics.
+//
+// - Endpoint: The endpoint to which metrics data will be exported. Expected format:
+//   - For gRPC: `grpc://host:port`
+//   - For HTTP: `http://host:port` or `https://host:port`
+//   - For file exporter: `file:///absolute/path/to/folder/file.json`
+type OpenTelemetryMetricsConfig struct {
+	Endpoint string
+}
+
+var (
+	otelInstance    *OpenTelemetry
+	otelConfig      *OpenTelemetryConfig
+	otelInitialized bool = false
+)
+
+// OpenTelemetry provides functionality for OpenTelemetry integration.
+// It can only be initialized once per process. Calling Init() more than once will be ignored.
+// If you need to change configuration, restart the process with new settings.
+type OpenTelemetry struct{}
+
+// GetInstance returns the singleton OpenTelemetry instance.
+func GetInstance() *OpenTelemetry {
+	if otelInstance == nil {
+		otelInstance = &OpenTelemetry{}
+	}
+	return otelInstance
+}
+
+// Init initializes the OpenTelemetry instance with the provided configuration.
+// It can only be called once per process. Subsequent calls will be ignored.
+func (o *OpenTelemetry) Init(openTelemetryConfig OpenTelemetryConfig) error {
```

Comment:
User can't access it, because it is in `internal`. You probably need to move a part outside.

Reply:
And add examples